CMake/RealmCore.cmake

```diff
@@ -272,22 +287,51 @@ endfunction()
 
 macro(build_realm_sync)
     set(cmake_files ${CMAKE_CURRENT_BINARY_DIR}${CMAKE_FILES_DIRECTORY})
-    ExternalProject_Add(realm-sync-lib
-        DEPENDS realm-core
-        PREFIX ${cmake_files}/realm-sync
-        BUILD_IN_SOURCE 1
-        BUILD_COMMAND make -C src/realm librealm-sync.a librealm-sync-dbg.a librealm-server.a librealm-server-dbg.a ${MAKE_FLAGS}
-        CONFIGURE_COMMAND ""
-        INSTALL_COMMAND ""
-        ${USES_TERMINAL_BUILD}
-        ${ARGV}
-        )
+    if(REALM_PLATFORM STREQUAL "Android")
+        ExternalProject_Add(realm-sync-lib
+            DEPENDS realm-core
+            PREFIX ${cmake_files}/realm-sync
+            BUILD_IN_SOURCE 1
+            BUILD_COMMAND sh build.sh build-android
```

Comment:
Ditto about hoisting the different command into a variable rather than repeating the entire `ExternalProject_Add`.